### ISSUE TITLE
Update Software Architecture Diagram

### ISSUE THREAD

**Issue by Nahrain1** [amosproj/amos2024ss07-updating-flash-boot-loader#72]
## User story
1. As a Software Developer
2. I want to update the software architecture diagram
3. So that it provides an up-to-date reference for the team regarding ongoing and future development activities

## Acceptance criteria
* The SW Architecure Diagram must be updated based on recent changes in the codebase
* The SW Architecure Diagram must reflect the functions used in the code
* All components and connections are clearly marked 

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
